tests/components/tessie/test_coordinator.py

```diff
@@ -27,81 +28,84 @@
 WAIT = timedelta(seconds=TESSIE_SYNC_INTERVAL)
 
 
-async def test_coordinator_online(
-    hass: HomeAssistant, mock_get_state, mock_get_status, freezer: FrozenDateTimeFactory
-) -> None:
-    """Tests that the coordinator handles online vehicles."""
+async def test_coordinator(hass: HomeAssistant, freezer: FrozenDateTimeFactory) -> None:
+    """Tests that the coordinator updates vehicles."""
 
-    await setup_platform(hass, PLATFORMS)
+    await setup_platform(hass, [Platform.BINARY_SENSOR])
 
     freezer.tick(WAIT)
     async_fire_time_changed(hass)
     await hass.async_block_till_done()
-    mock_get_status.assert_called_once()
-    mock_get_state.assert_called_once()
-    assert hass.states.get("binary_sensor.test_status").state == STATE_ON
+
+    # Since the fixture data is old, the status will be offline
+    assert hass.states.get("binary_sensor.test_status").state == STATE_OFF
 
 
-async def test_coordinator_asleep(
-    hass: HomeAssistant, mock_get_status, freezer: FrozenDateTimeFactory
+async def test_coordinator_online(
+    hass: HomeAssistant, freezer: FrozenDateTimeFactory, mock_get_state: AsyncMock
 ) -> None:
-    """Tests that the coordinator handles asleep vehicles."""
+    """Tests that the coordinator updates vehicles with recent data."""
 
     await setup_platform(hass, [Platform.BINARY_SENSOR])
-    mock_get_status.return_value = TEST_VEHICLE_STATUS_ASLEEP
 
-    freezer.tick(WAIT)
+    # Update the fixture to be "current"
+    future = datetime.now()
+    data = deepcopy(TEST_VEHICLE_STATE_ONLINE)
+    data["vehicle_state"]["timestamp"] = int(future.timestamp() * 1000)
+    mock_get_state.return_value = data
+
+    freezer.move_to(future)
     async_fire_time_changed(hass)
     await hass.async_block_till_done()
-    mock_get_status.assert_called_once()
-    assert hass.states.get("binary_sensor.test_status").state == STATE_OFF
+
+    # Since the timestamp is now current, the state will be online
+    assert hass.states.get("binary_sensor.test_status").state == STATE_ON
 
 
 async def test_coordinator_clienterror(
-    hass: HomeAssistant, mock_get_status, freezer: FrozenDateTimeFactory
+    hass: HomeAssistant, mock_get_state: AsyncMock, freezer: FrozenDateTimeFactory
 ) -> None:
     """Tests that the coordinator handles client errors."""
 
-    mock_get_status.side_effect = ERROR_UNKNOWN
+    mock_get_state.side_effect = ERROR_UNKNOWN
     await setup_platform(hass, [Platform.BINARY_SENSOR])
 
     freezer.tick(WAIT)
     async_fire_time_changed(hass)
     await hass.async_block_till_done()
-    mock_get_status.assert_called_once()
```

Comment:
Should we not keep it but for `get_state` instead?
Same below I see.

Reply:
Yes